AppCenterDataStorage/AppCenterDataStorageTests/MSDBDocumentStoreTests.m

```diff
@@ -202,6 +200,29 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
   XCTAssertEqualObjects(documentWrapper.documentId, documentId);
 }
 
+- (void)testUpsertReplacesCorrectlyInAppStorage {
```

Comment:
Test case suggestion:

On Android we have a test case (https://github.com/Microsoft/AppCenter-SDK-Android/blob/b563fc68e6919d2a20dd12fd367ca4c080b1f7f9/sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/DatabaseManagerAndroidTest.java#L498) that verifies `replace` called twice with insertion the first time and update the second time. 

I haven't seen any other test methods in this class verifying `replace` called only once, so maybe we do the same test as Android? 😄